cmd/compose/up.go

```diff
@@ -170,6 +171,7 @@ func upCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service) *c
 	flags.BoolVar(&up.wait, "wait", false, "Wait for services to be running|healthy. Implies detached mode.")
 	flags.IntVar(&up.waitTimeout, "wait-timeout", 0, "Maximum duration in seconds to wait for the project to be running|healthy")
 	flags.BoolVarP(&up.watch, "watch", "w", false, "Watch source code and rebuild/refresh containers when files are updated.")
+	flags.BoolVar(&up.prune, "prune", false, "Prune dangling images on rebuild")
```

Comment:
I wonder: is there any benefits _not_ pruning those ? Shall we just make this the default ? Then do we need a flag ?

Reply:
Yep, I agree
@remcokranenburg do you mind adapting your PR to prune by default? We just need to keep it for the `watch` command itself to let users who want it to choose the option of keeping dangling images

Reply:
up already has so much flags we'd prefer not to have more. For those who really want to keep dangling images (?) `watch --prune=false` could be a reasonable workaround

Reply:
Yes I'm aligned with @ndeloof on this one, just keep the `--prune` flag for the `watch` command
BTW thank you for doing it 🙏